tests/test_graphql_api/test_application_event_schedule/test_approve.py

```diff
@@ -184,3 +205,331 @@ def test_cannot_approve_if_application_is_not_in_unallocated(graphql):
+
+
+def test_approve_schedule__approved_time_falls_on_two_separated_wished_times(graphql):
+    # given:
+    # - There is an allocatable application event schedule
+    # - A superuser is using the system
+    application = ApplicationFactory.create_application_ready_for_allocation()
+    graphql.login_user_based_on_type(UserType.SUPERUSER)
+
+    ApplicationEventScheduleFactory.create(
+        application_event=application.application_events.first(),
+        day=WeekdayChoice.MONDAY,
+        begin=datetime.time(14, 30, tzinfo=get_default_timezone()),
+        end=datetime.time(16, 0, tzinfo=get_default_timezone()),
+    )
+
+    # when:
+    # - The user tries to allocate the schedule, and it falls
+    #   on two of the applicants the wished periods.
+    input_data = approve_data(application, begin="13:00:00", end="15:00:00")
+    response = graphql(APPROVE_MUTATION, input_data=input_data)
+
+    # then:
+    # - The response complains about the allocated time being invalid.
+    assert response.field_error_messages() == [
+        "Cannot allocate schedule for this day and time period. "
+        "Given time period does not fit within applicants wished periods.",
+    ]
+
+
+def test_approve_schedule__cannot_approve_to_reservation_unit_not_in_event_reservation_units(graphql):
+    # given:
+    # - There is an allocatable application event schedule
+    # - A superuser is using the system
+    application = ApplicationFactory.create_application_ready_for_allocation()
+    graphql.login_user_based_on_type(UserType.SUPERUSER)
+
+    # when:
+    # - The user tries to allocate the schedule, but for a reservation unit
+    #   not in the event's event reservation units.
+    input_data = approve_data(application, begin="10:00:00", end="12:00:00")
+    input_data["allocatedReservationUnit"] = ReservationUnitFactory.create().pk
+    response = graphql(APPROVE_MUTATION, input_data=input_data)
+
+    # then:
+    # - The response complains about the reservation unit not being included.
+    assert response.field_error_messages("allocatedReservationUnit") == [
+        "Cannot allocate schedule for this reservation unit. Reservation unit is not included in the event.",
+    ]
+
+
+def test_approve_schedule__cannot_approve_two_schedules_for_same_day(graphql):
+    # given:
+    # - There is an allocatable application event schedule
+    # - A superuser is using the system
+    application = ApplicationFactory.create_application_ready_for_allocation(events_per_week=3, pre_allocated=True)
+    graphql.login_user_based_on_type(UserType.SUPERUSER)
+
+    ApplicationEventScheduleFactory.create(
+        day=WeekdayChoice.MONDAY,
+        application_event=application.application_events.first(),
+    )
+
+    # when:
+    # - The user tries to allocate the schedule, but for a reservation unit
+    #   not in the event's event reservation units.
+    input_data = approve_data(application, begin="12:00:00", end="14:00:00")
```

Comment:
Could test with the force flag to prove that it does not allow overriding this check.

Reply:
Done